sklearn/decomposition/online_lda.py

```diff
@@ -250,6 +250,35 @@ class LatentDirichletAllocation(BaseEstimator, TransformerMixin):
     n_iter_ : int
         Number of passes over the dataset.
 
+    Examples
+    --------
+    >>> from sklearn.feature_extraction.text import CountVectorizer
+    >>> from sklearn.decomposition import LatentDirichletAllocation
+    >>> from sklearn.datasets import fetch_20newsgroups
+    >>> n_samples = 1000
+    >>> n_features = 200
+    >>> n_components = 5
+    >>> dataset = fetch_20newsgroups(shuffle=True, random_state=1,
+    ...                              remove=('headers', 'footers', 'quotes'))
+    >>> X = dataset.data[:n_samples]
+    >>> tf_vectorizer = CountVectorizer(max_features=n_features,
+    ...                                 stop_words='english')
+    >>> tf = tf_vectorizer.fit_transform(X)
+    >>> lda = LatentDirichletAllocation(n_components=n_components,
+    ...     random_state=0)
+    >>> lda.fit(tf)
+    LatentDirichletAllocation(batch_size=128, doc_topic_prior=None,
+                 evaluate_every=-1, learning_decay=0.7,
+                 learning_method='batch', learning_offset=10.0,
+                 max_doc_update_iter=100, max_iter=10, mean_change_tol=0.001,
+                 n_components=5, n_jobs=None, n_topics=None, perp_tol=0.1,
+                 random_state=0, topic_word_prior=None,
+                 total_samples=1000000.0, verbose=0)
+    >>> # get topics for some given samples:
+    >>> lda.transform(tf_vectorizer.transform(dataset.data[-2:]))
```

Comment:
The first issue, is that this takes 6s to run on my laptop, it will probably take more in CI. That is somewhat slow for a doctest. Using `fetch_20newsgroups_vectorized` together with only a few classes, might help.

The second issue is that 20 newsgroups is not a mandatory dataset. Currently, tests and doctests that require it are optionally skipped if it is not present (see `setup_twenty_newsgroups` in `doc/conftest.py`) and `sklearn/datasets/tests/test_20news.py`. We might use a similar approach (if possible), though I'm not sure this example would justify the effort. Maybe we could use just some random sparse dataset?

Reply:
Ok, it's changed to `make_multilabel_classification` now. But I'd really rather have something which takes text as the input data, and not count vectorized data. It kinda makes it not clear that it's count vectorized (I assume it is), and a lot of processing that goes on processing a text dataset is hidden somehow there.